OptimizelyDemoApp/AppDelegate.swift

```diff
@@ -43,6 +46,75 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
     
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {
         
+        // ********************************************************
+        // ***************** Integration Samples ******************
+        // ********************************************************
+        
+        // most of the third-party integrations only support iOS, so the samples will only work for iOS builds
+#if os(iOS)
+    let defaultNotificationCenter = NotificationCenter.default
+    defaultNotificationCenter.addObserver(forName: NSNotification.Name("OptimizelyExperimentActivated"), object: nil, queue: nil) { (note) in
+        
+        print("Received an activation notification: \n", note)
+        
+        let userInfo : Dictionary<String, AnyObject>? = note.userInfo as! Dictionary<String, AnyObject>?
+        if let experiment = userInfo?["experiment"] as! OPTLYExperiment? {
+            if let variation = userInfo?["variation"] as! OPTLYVariation? {
+                
+                // ---- Amplitude ----
+                let propertyKey : String! = "[Optimizely] " + experiment.experimentKey
+                let identify : AMPIdentify = AMPIdentify()
+                identify.set(propertyKey, value:variation.variationKey as NSObject!)
+                
+                // Track impression event (optional)
+                let eventIdentifier : String = "[Optimizely] " + experiment.experimentKey + " " + variation.variationKey
+                Amplitude.instance().logEvent(eventIdentifier)
+                
+                // ---- Google Analytics ----
+                let tracker : GAITracker? = GAI.sharedInstance().defaultTracker
```

Comment:
no they're integration is weird -- i had to create a bridging header.
i actually was disappointed how hard it was to get GA in the app.
they also don't support tvOS.